### PR TITLE
Introduce flutter-warmed image

### LINKED ISSUE #10: Warmed up flutter for web
I wonder if you would be ok to introduce `flutter-warmed` wnich would prepare `flutter` image to web development by running:

```
flutter upgrade
flutter precache --no-ios --no-linux --no-windows --no-winuwp --no-macos --no-fuchsia --universal --web
```

This will be a good option for user to have flutter ready from start instead of waiting for image to build and download related tools
It will relatively fatten up docker image but this is expected in the same way as for android variant.

It is not a big deal though as I can prepare my own image based on yours, but I just wanted to suggest it as convenience

### PR DESCRIPTION
**Pull request by DoumanAsh**
Fixes #10

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,6 +19,10 @@ ifdef FLUTTER_CHANNEL
 		 --file ./dockerfiles/flutter.dockerfile \
 		 --build-arg FLUTTER_CHANNEL=$(FLUTTER_CHANNEL) \
 		 --tag plugfox/flutter:$(FLUTTER_CHANNEL) .
+	docker build --no-cache --force-rm --compress \
+		 --file ./dockerfiles/flutter_warmed.dockerfile \
+		 --build-arg FLUTTER_CHANNEL=$(FLUTTER_CHANNEL) \
+		 --tag plugfox/flutter:$(FLUTTER_CHANNEL)-warmed .
```
Comment (PlugFox) [inline Makefile:25]: plugfox/flutter:$(FLUTTER_CHANNEL)-web-warmed

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,10 @@ ifdef FLUTTER_CHANNEL
 		 --file ./dockerfiles/flutter.dockerfile \
 		 --build-arg FLUTTER_CHANNEL=$(FLUTTER_CHANNEL) \
 		 --tag plugfox/flutter:$(FLUTTER_CHANNEL) .
+	docker build --no-cache --force-rm --compress \
+		 --file ./dockerfiles/flutter_warmed.dockerfile \
```
Comment (PlugFox) [inline Makefile:23]: flutter_web_warmed.dockerfile

File: Makefile [hunk 3]
```diff
@@ -34,6 +38,10 @@ ifdef FLUTTER_VERSION
 		 --file ./dockerfiles/flutter.dockerfile \
 		 --build-arg FLUTTER_VERSION=$(FLUTTER_VERSION) \
 		 --tag plugfox/flutter:$(FLUTTER_VERSION) .
+	docker build --no-cache --force-rm --compress \
+		 --file ./dockerfiles/flutter_warmed.dockerfile \
```
Comment (PlugFox) [inline Makefile:42]: flutter_web_warmed.dockerfile

File: Makefile [hunk 4]
```diff
@@ -34,6 +38,10 @@ ifdef FLUTTER_VERSION
...
+		 --build-arg FLUTTER_VERSION=$(FLUTTER_VERSION) \
+		 --tag plugfox/flutter:$(FLUTTER_VERSION)-warmed .
```
Comment (PlugFox) [inline Makefile:44]: plugfox/flutter:$(FLUTTER_VERSION)-web-warmed

File: Makefile [hunk 5]
```diff
@@ -52,12 +60,14 @@ push:
...
 	@echo "PUSH FLUTTER $(FLUTTER_CHANNEL)"
 	docker push plugfox/flutter:$(FLUTTER_CHANNEL)
+	docker push plugfox/flutter:$(FLUTTER_CHANNEL)-warmed
 	docker push plugfox/flutter:$(FLUTTER_CHANNEL)-android
 	docker push plugfox/flutter:$(FLUTTER_CHANNEL)-android-warmed
 endif
 ifdef FLUTTER_VERSION
 	@echo "PUSH FLUTTER $(FLUTTER_VERSION)"
 	docker push plugfox/flutter:$(FLUTTER_VERSION)
+	docker push plugfox/flutter:$(FLUTTER_VERSION)-warmed
```
Comment (PlugFox) [inline Makefile:70]: plugfox/flutter:$(FLUTTER_VERSION)-web-warmed

File: Makefile [hunk 6]
```diff
@@ -52,12 +60,14 @@ push:
 ifdef FLUTTER_CHANNEL
 	@echo "PUSH FLUTTER $(FLUTTER_CHANNEL)"
 	docker push plugfox/flutter:$(FLUTTER_CHANNEL)
+	docker push plugfox/flutter:$(FLUTTER_CHANNEL)-warmed
```
Comment (PlugFox) [inline Makefile:63]: plugfox/flutter:$(FLUTTER_CHANNEL)-web-warmed

File: dockerfiles/flutter_warmed.dockerfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+FROM plugfox/flutter:${FLUTTER_CHANNEL}${FLUTTER_VERSION}
+
+# Setup flutter tools for web developement
+# Unfortunately flutter doesn't have convenient way to tell it to install web/universal only
+RUN set -eux; flutter --suppress-analytics upgrade; \
+    flutter --suppress-analytics precache --no-ios --no-linux --no-windows --no-winuwp --no-macos --no-fuchsia --universal --web
+
+# Add lables
+LABEL name="plugfox/flutter-warmed:${FLUTTER_CHANNEL}${FLUTTER_VERSION}" \
```
Comment (PlugFox) [inline dockerfiles/flutter_warmed.dockerfile:12]: name="plugfox/flutter:${FLUTTER_CHANNEL}${FLUTTER_VERSION}"-web-warmed

File: dockerfiles/flutter_warmed.dockerfile [hunk 8]
```diff
@@ -0,0 +1,31 @@
+ARG FLUTTER_CHANNEL=""
+ARG FLUTTER_VERSION=""
+
+FROM plugfox/flutter:${FLUTTER_CHANNEL}${FLUTTER_VERSION}
+
+# Setup flutter tools for web developement
+# Unfortunately flutter doesn't have convenient way to tell it to install web/universal only
+RUN set -eux; flutter --suppress-analytics upgrade; \
+    flutter --suppress-analytics precache --no-ios --no-linux --no-windows --no-winuwp --no-macos --no-fuchsia --universal --web
+
```
Comment (PlugFox) [inline dockerfiles/flutter_warmed.dockerfile:10]: ... \
&& flutter config --no-analytics

File: dockerfiles/flutter_warmed.dockerfile [hunk 9]
```diff
@@ -0,0 +1,31 @@
+ARG FLUTTER_CHANNEL=""
+ARG FLUTTER_VERSION=""
+
+FROM plugfox/flutter:${FLUTTER_CHANNEL}${FLUTTER_VERSION}
+
```
Comment (PlugFox) [inline dockerfiles/flutter_warmed.dockerfile:5]: ARG FLUTTER_CHANNEL
ARG FLUTTER_VERSION


WORKDIR /

**Comment by DoumanAsh** [comment]
I've applied comments.
Btw final image size is 1.35GB

**Comment by PlugFox** [comment]
I'll see later

**Comment by DoumanAsh** [comment]
I was testing my porject build, it still downloads these `universal` tools when you do `pub get` in project.
So probably there is no way to avoid needing it

**Comment by PlugFox** [comment]
@DoumanAsh have you any additional changes?
Can I proceed with the current Pull Request and merge it?

**Comment by DoumanAsh** [comment]
@PlugFox No, I believe there is nothing else to add, it should be perfectly ready

**Pull request merged**